tensorflow/contrib/lite/kernels/ssd_ops.cc

```diff
@@ -0,0 +1,1184 @@
+#include "tensorflow/contrib/lite/kernels/ssd_ops.h"
+
+#include <chrono>
+#include <iostream>
+#include <tuple>
+#include <vector>
+
+#include "tensorflow/contrib/lite/context.h"
+#include "tensorflow/contrib/lite/kernels/internal/tensor.h"
+#include "tensorflow/contrib/lite/kernels/internal/types.h"
+#include "tensorflow/contrib/lite/kernels/kernel_util.h"
+#include "tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h"
+#include "tensorflow/contrib/lite/kernels/internal/reference/reference_ops.h"
+
+namespace tflite {
+namespace ops {
+namespace custom {
+
+
+inline void Exp(const float* input_data, const Dims<4>& input_dims,
+                float* output_data, const Dims<4>& output_dims) {
+  // Get sizes
+  const int batches = MatchingArraySize(input_dims, 3, output_dims, 3);
+  const int height = MatchingArraySize(input_dims, 2, output_dims, 2);
+  const int width = MatchingArraySize(input_dims, 1, output_dims, 1);
+  const int depth = MatchingArraySize(input_dims, 0, output_dims, 0);
+
+  // Loop
+  for (int b = 0; b < batches; ++b) {
+    for (int y = 0; y < height; ++y) {
+      for (int x = 0; x < width; ++x) {
+        for (int c = 0; c < depth; ++c) {
+          // Get offsets
+          int input_offset = Offset(input_dims, c, x, y, b);
+          int output_offset = Offset(output_dims, c, x, y, b);
+
+          // Compute exp
+          output_data[output_offset] = std::exp(input_data[input_offset]);
```

Comment:
Perhaps a naive question, but why do you write inefficient code like this when we have SIMD vectorized kernels for the same ops in TensorFlow (based on Eigen)?

Reply:
As some background, I asked Mohamed to share his current code as a PR so that we could understand what the current state is, what ops it covers, and collaborate on merging. So I'm expecting that we'll be working together on this to make changes before we expect it to be added to master.

Reply:
@rmlarsen You are right. This code is just a reference implementation to check correctness of further improvements. And I will be working with @petewarden and you guys to optimize it.